packages/core/src/client.ts

```diff
@@ -360,6 +360,10 @@ export interface MedplumRequestOptions extends RequestInit {
    */
   pollStatusPeriod?: number;
 
+  /**
+   * Optional AbortSignal for cancelling an async request.
+   */
+  asyncReqCancelSignal?: AbortSignal;
```

Comment:
Can we just use the normal `options.signal`?

Reply:
Nope because that will abort the fetch request itself as well. I originally tried that but it was hard to untangle the options since you have to, in the middle of a bunch of tied together requests, remove the signal from the options or else risk other fetch requests made in the lifecycle of the parent request being pre-aborted. 

Logically these are two separate signals as well, so it kind of makes sense not to overload them. Maybe there's a complicated way but it seemed to preserve the intent of the signal better to separate it out.

Reply:
Ok I tried to make it work again and I remember why it didn't work last time. 

We have to pass through the signal to each subsequent call to `pollStatus`, which means if you pass the signal through to `request`, it will be passed to `fetch` and if abort the signal while that request is still in flight.

You don't get the next call to `pollStatus` where it would have picked up that it needs to cancel and so it gets stuck and never cancels, just stops polling.

Reply:
Ok, as discussed offline did a big refactor that now only touches `packages/app` for this part. Cancelling done via `DELETE` on status URL and then just going down the same server side path letting the cancel pop up naturally while polling